### PR TITLE
feat(web): surface the tips system on the dashboard as a tip-of-the-day

### LINKED ISSUE #2292: Surface the tips system on the web dashboard
_Note: this issue was drafted by Claude via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

## Background

The tips system landed for the TUI in #2282 (which closes #2262). The core engine lives in the shared lib at `src/tips.rs`: a render-free `Tip { id, title, body, trigger }` catalog with `Rotation` / `Earned` triggers plus the eligibility, unseen-count, and next-pop selection logic. Seen and dismissed state lives in `app_state` (`tips_disabled`, `tips_seen`, `new_session_with_selection_count`), which the `aoe serve` server already reads and writes, so it is shared across surfaces.

It was deliberately built so the web dashboard can reuse the same catalog with no rework. This issue tracks that follow-up.

## Scope

Surface tips on the web dashboard, mirroring the TUI:

- **Server**: expose the catalog and eligibility for the web surface, e.g. `GET /api/tips` returning the eligible / unseen tips (the template is the settings-schema endpoint, `GET /api/settings/schema`). Add endpoints to mark a tip seen and to toggle "don't show tips" (mirror `POST /api/app-state/web-tour-seen`). Reuse `crate::tips` and `app_state` so persistence stays shared with the TUI.
- **Web UI**: a small `💡 N` badge (model on `TopBar`) and a tips panel or modal (model on `TelemetryConsentModal` / `SessionGroupModal`), reading the API. Mark-seen-on-view plus don't-show-again, matching the TUI overlay.
- **Surface tagging**: introduce per-tip surface applicability (TUI-only, web-only, or both) so keyboard-only tips like the `N` / Shift+N shortcut are not shown on the web, and a future web-only tip (e.g. "install the dashboard as a PWA") can be web-only. The current `N` earned tip is TUI-only.

## Notes / requirements

- No dead code: this is why the server endpoints and surface tagging were not added in #2282. Nothing would have consumed them yet, and clippy plus the repo's no-dead-code rule would reject them.
- Per AGENTS.md, this touches a user-facing dashboard flow, so it needs a `web/tests/coverage-matrix.json` entry and Playwright coverage (mocked plus live as appropriate), plus the usual web checks (oxfmt, ESLint, tsc).
- Earned-tip triggers that depend on TUI-specific behavior (the new-session-with-selection counter) may not map cleanly to the web; decide per tip whether an earned tip is cross-surface or surface-specific.

## Out of scope

The TUI tips system itself (shipped in #2282).

### PR DESCRIPTION
**Pull request by Seluj78**
**Note:** Claude handled the implementation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

## Description

Surfaces the tips system on the web dashboard, mirroring the TUI (the follow-up tracked by #2292 after the TUI tips landed in #2282), and populates the catalog with tips for the dashboard's features.

The shared `crate::tips` engine is now surface-aware: each `Tip` lists the surfaces it applies to (TUI and/or Web) and every eligibility query takes a `TipSurface`, so keyboard-only hints stay out of the web and web-only hints stay out of the TUI.

On the web the tips render as a GIMP/DBeaver-style **tip-of-the-day modal**: one tip at a time with Previous / "Tip X of N" / Next, a "Show tips on startup" checkbox, and Close. It auto-pops once on startup when a tip is unseen and tips are on (gated behind first-run onboarding and suppressed in automated browser sessions so it can't intercept the rest of the suite), and is reopenable any time from the top-bar overflow menu (More options > Tips). Each tip is marked seen as it is shown; seen and the on/off preference persist server-side, shared with the TUI across devices.

Server endpoints (all read projections / single-purpose writes off the elevation wall, since `PATCH /api/settings` is passphrase-gated): `GET /api/tips` returns the web-surface tips with `enabled` and per-tip `seen`; `POST /api/app-state/tip-seen {id}` marks one seen (rejecting ids not in the catalog); `POST /api/tips/show {enabled}` is the "Show tips on startup" toggle.

Catalog: seven web feature-discovery tips (pin, archive, snooze, grouping, sorting, scratch sessions, multi-repo) written in web-gesture language, plus three grouped TUI keyboard-shortcut tips (core views, triage, power moves) whose key labels are resolved live so they stay correct in strict-hotkey mode. All rotation. No migration; the new rotation tips light the badge/modal once as a feature announcement, opt-out via Show tips.

Closes #2292

## PR Type

- [x] New Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation
- [ ] Infrastructure / CI

## How to test

- [ ] Run `aoe serve` on a fresh config. Open More options > Tips; confirm the "Tip of the day" modal opens and steps through the web tips with Next/Previous.
- [ ] Confirm the modal shows web tips (Install as an app, pin, archive, snooze, grouping, sorting, scratch, multi-repo) and never the TUI-only "Reuse the selected session's settings" tip.
- [ ] Read a tip, close, reload; confirm it does not re-show that tip (seen persists server-side; `GET /api/tips` reports `seen: true`).
- [ ] Uncheck "Show tips on startup"; reload and confirm it no longer auto-pops, and Settings > Interaction > Show tips is now off. Re-check it (or the setting) to turn it back on.
- [ ] On a fresh browser, get past the theme welcome + tour; confirm the tip-of-the-day auto-pops once on the first unseen tip.
- [ ] In the TUI, confirm the `N` / new-from-selection tip still earns and pops, and the new keyboard-shortcut tips show the correct keys (including in strict-hotkey mode).

## Checklist

- [x] I understand the code I am submitting
- [x] New and existing tests pass
- [x] Documentation was updated where necessary
- [ ] For UI changes: included screenshot or recording

## Test Coverage Analysis

**Web dashboard / structured view (Playwright user story)**
- [x] Added or updated a Playwright spec under `web/tests/` and updated `web/tests/coverage-matrix.json`

Live Playwright (`web/tests/live/tips.spec.ts`) covers the startup auto-pop marking the shown tip seen (persisted, no TUI-tip leak) and the menu-reopen path with the "Show tips on startup" toggle persisting. Vitest covers the TipsModal carousel (start index, Next/Previous + wrap, mark-seen on navigate, the checkbox, single-tip and empty states) and the TopBar "Tips" menu entry. Rust unit tests cover surface filtering, web-tips-carry-no-placeholders, the keybinding placeholder resolver, and id validation; the server read-only-guard audit covers the new write handlers.

**TUI / CLI (e2e test)**
- [x] N/A: this PR doesn't change TUI rendering, a CLI subcommand, or session lifecycle

The TUI gains three rotation tips and a generalized placeholder resolver; both are covered by the engine and home-view unit tests (the existing tips/overlay/home suites were extended).

## AI Usage

- [ ] No AI was used
- [x] AI was used

**AI Model/Tool used:** Claude (via Claude Code, agent-of-empires `/aoe-implement` skill)

**Any Additional AI Details you'd like to share:**

Investigation, plan (with a multi-model `/debate` consult), and implementation drafted by Claude under my direction. I made the design calls, including pivoting from an initial badge to the tip-of-the-day modal and overriding the debate's "reuse PATCH /api/settings" verdict once it surfaced that path is elevation-gated. I reviewed each commit and ran the manual test steps.

- [x] I am an AI Agent filling out this form (check box if true)

---
<a href="https://app.blacksmith.sh/agent-of-empires/codesmith/agent-of-empires/pr/2332"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1784731700&installation_id=139138837&pr_number=2332&repository=agent-of-empires%2Fagent-of-empires&return_to=https%3A%2F%2Fgithub.com%2Fagent-of-empires%2Fagent-of-empires%2Fpull%2F2332&signature=7c54fa66790a59570628e2e09c9def29e7aca999698c310d0ee482cdcf63c7ca"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## Summary

This PR brings the existing “tips of the day” system to the web dashboard. It reuses the same shared Rust tips engine that powers the TUI, adds “where this tip should appear” rules (Web vs TUI), and then renders the web version as a one-tip-at-a-time modal with navigation, onboarding-safe auto-pop, and user-controlled “show on startup” + “seen” persistence.

### Shared tips engine (Rust)
- Added **per-tip surface targeting** so Web-only tips don’t show up in the TUI, and TUI keyboard-shortcut tips don’t leak into the web.
- Updated the eligibility API to require a **target surface** (Web or TUI).
- Expanded the tip catalog with **Web feature-discovery tips** (multiple app/product discovery actions).
- Added **tip id validation** before marking tips as seen.
- Updated TUI tip rendering to **substitute all supported shortcut placeholders** (so keyboard labels resolve correctly in strict mode).

### Server API (web dashboard)
- Added **`GET /api/tips`** to return Web-eligible tips, including:
  - whether tips are enabled
  - per-tip `seen` status
- Added **`POST /api/app-state/tip-seen`** to mark a tip as seen (server persisted).
- Added **`POST /api/tips/show`** to toggle **“Show tips on startup”** (server persisted).
- All persistence is shared across web and TUI so the same user/tip state carries across devices.

### Web UI (React)
- Added a **Tip of the day modal** with:
  - one tip at a time
  - Previous/Next navigation (wraps)
  - Close + Escape handling
  - “Show tips on startup” checkbox
  - auto-marking tips as seen when shown/navigated to
- Implemented startup behavior to **auto-open once on page load** when conditions are met and suppresses auto-popup in automated browser sessions.
- Rewired modal opening into the **top-bar overflow menu**: **More options → Tips**.
- Refactored tips modal orchestration into a `useTips()` hook using a dedicated startup gate (`shouldAutoPopTips`).

## Testing
- Added **Vitest + React Testing Library** tests for the modal (navigation, close paths, checkbox behavior, edge cases).
- Added **unit tests** for `useTips()` and the startup gate logic.
- Added **Playwright live tests** validating:
  - correct Web-only tip selection vs TUI-only tip suppression
  - persistence of `seen` and enabled/disabled settings across reloads
  - the tutorial flow explicitly disables tips so the tutorial modal isn’t competing.
- Updated `web/tests/coverage-matrix.json` to include new coverage areas.

## Benefits
- **Consistent cross-platform onboarding**: one shared tips system with shared persistence across web and TUI.
- **Correct targeting**: surface tagging prevents the wrong style of tips (e.g., keyboard-shortcuts) from appearing on the web.
- **Better user control**: users can disable “show on startup,” close the modal, and reopen it later from the menu.

### CODE REVIEW

File: docs/guides/web-dashboard.md [hunk 1]
```diff
@@ -180,4 +180,10 @@ When you leave the PWA and come back, it reopens to the session you last had ope
 
 `Ctrl-C` on a foreground server, or `aoe serve --stop` against a daemon, both exit within ~5 seconds even with open tabs. Live clients receive a `1001` ("going away") close frame and reconnect once a fresh server is running.
 
+## Tips
+
+The dashboard surfaces the same "did you know" tips as the TUI, as a tip-of-the-day modal. When you have an unseen tip, it pops once on startup (after the first-run onboarding settles); step through tips with Previous and Next. Reopen it any time from the top-bar menu (More options > Tips). Each tip is marked read as you view it, and read state is stored on the server, so it follows you across browsers and devices and a tip never re-pops once seen.
+
+The "Show tips on startup" checkbox in the modal controls whether it auto-pops; it sets the same `Show tips` preference as the TUI, also editable under Settings > Interaction > Show tips. Tips are surface-aware: keyboard-shortcut tips stay in the TUI and never show on the dashboard.
+
```
Comment (njbrake) [inline docs/guides/web-dashboard.md]: @Seluj78 this seems unnecessary to be putting into a guide, WDYT?

Comment (Seluj78) [reply docs/guides/web-dashboard.md]: Agreed, the modal is self-explanatory. Dropped the guide section (and its intro-line mention) in bb4efd5c.

**Comment by njbrake** [comment]
@Seluj78 a bit higher level question: is there some re-architecting we should do for the tips interface so that you don't need to mirror logic? I was surprised about how much new rust code was added in this PR, I had tried to implement the tips system so that it would be easy for you to utilize it for the web, maybe I didn't quite get it right in my first attempt?

**Comment by Seluj78** [comment]
> **Note:** Claude handled the investigation and prose via back-and-forth. Idea, decisions, and everything else are mine. — @Seluj78

Good question, and the good news is the engine reuse worked as you intended: the web doesn't mirror any eligibility logic. `GET /api/tips` calls `crate::tips::eligible(Web, signals)` directly and the React side is dumb. Adding a future tip is one `CATALOG` entry with zero handler or frontend change.

On the volume of Rust, most of it isn't logic:

- `src/tips.rs` (+334): mostly data and tests. About 100 lines are the 10 tip entries themselves, about 150 are unit tests. The actual new logic is small: the `TipSurface` enum, the `surfaces` field, threading a surface arg through the four query fns, and `id_in_catalog`.
- `src/server/api/system.rs` (+193): the three endpoints (`GET /api/tips`, `tip-seen`, `tips/show`). They are long only because they follow the existing single-purpose write pattern (`mark_web_tour_seen`): read-only guard, `spawn_blocking`, and a three-arm error match each.

The one real change to your v1 was adding surface tagging. v1 had no surface concept, so a keyboard-shortcut tip (the `N` one) would have leaked onto the web; tagging keeps each surface to its own tips. It is a shared engine enhancement, not web-side duplication. The endpoints are what #2292 scoped, so this is the planned follow-up rather than a sign the first attempt was wrong.

The only real lever to cut Rust is the per-handler boilerplate: the app-state single-purpose writes (`web-tour-seen`, `dismiss-update`, `tip-seen`, `tips/show`) all repeat read-only guard + `spawn_blocking` + `Config` load/save. A small shared helper would shrink each to a few lines. Happy to do that as a separate cleanup PR if you want it; it is repo-wide, not tips-specific.

**Review by njbrake — APPROVED** [review]
_Note: this review was drafted by Claude via back-and-forth with @njbrake. The reasoning and decisions are his; the prose is Claude's._

Reviewed the implementation end to end and ran the suite locally against the branch: the Rust tips engine plus TUI dialog tests (24 passed), web Vitest for the tip modal, hook, and API client (33 passed), and tsc, ESLint, and oxfmt clean. The engine reuse landed the way you intended: the web side just calls `eligible(Web, ...)` and renders, so no eligibility logic is mirrored and adding a tip stays one `CATALOG` entry.

Pushed three small follow-ups in bb9948c7:

1. Gate the startup auto-pop on telemetry resolution. `telemetryConsentNeeded` defaults to false, so before the status fetch settled, "no consent needed" and "not resolved yet" looked identical, which let the tip modal slip in ahead of a consent modal the in-flight fetch was about to raise. It now waits for the fetch to settle. The live harness sets `DO_NOT_TRACK`, so `tips.spec.ts` still pops cleanly.
2. Corrected the stale `TipsResponse.enabled` doc: the toggle now writes through `POST /api/tips/show`, not the settings schema.
3. Asserted the tips-disable setup POST in the tutorial live spec so a setup failure fails fast instead of surfacing later as a confusing modal symptom.

One behavior change worth a conscious nod rather than a code edit: the shared catalog now ships three always-eligible TUI rotation tips, and since `show_tips` defaults on, existing TUI users will see a "3" badge after upgrading where a fresh user previously saw none. That is intentional and called out in your description; flagging it so the change is a deliberate choice.

Approving.
